projects/igniteui-angular/src/lib/date-common/util/date-time.util.ts

```diff
@@ -0,0 +1,525 @@
+import { DatePart, DatePartInfo } from '../../directives/date-time-editor/date-time-editor.common';
+import { formatDate, FormatWidth, getLocaleDateFormat } from '@angular/common';
+import { ValidationErrors } from '@angular/forms';
+import { isDate, parseDate } from '../../core/utils';
+
+/** @hidden */
+const enum FormatDesc {
+    Numeric = 'numeric',
+    TwoDigits = '2-digit'
+}
+
+const DATE_CHARS = ['h', 'H', 'm', 's', 'S', 't', 'T'];
+const TIME_CHARS = ['d', 'D', 'M', 'y', 'Y'];
+
+/** @hidden */
+const enum DateParts {
+    Day = 'day',
+    Month = 'month',
+    Year = 'year'
+}
+
+/** @hidden */
+export class DateTimeUtil {
```

Comment:
Not that I'm worried for an internal class, but being marked as `abstract` was a valid valid way to keep the class from being instantiated :)